diff --git a/lib/theme/ApiExplorer/CodeTabs/index.js b/lib/theme/ApiExplorer/CodeTabs/index.js
index a0e09d4c5d32af8d1792ebe2c6eb63f7872f5c15..992609f36db1894d179c3ee23d42a8acf42ea972 100644
--- a/lib/theme/ApiExplorer/CodeTabs/index.js
+++ b/lib/theme/ApiExplorer/CodeTabs/index.js
@@ -240,7 +240,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   const { className } = props;
   return react_1.default.createElement(
     "div",
diff --git a/lib/theme/ApiTabs/index.js b/lib/theme/ApiTabs/index.js
index 660f4905897a5904085a9192ba1efa639d67d0a7..7e0ecf806423e7124a1be21dc9cdf8f2b3156afb 100644
--- a/lib/theme/ApiTabs/index.js
+++ b/lib/theme/ApiTabs/index.js
@@ -250,7 +250,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   return react_1.default.createElement(
     "div",
     { className: "openapi-tabs__container" },
diff --git a/lib/theme/DiscriminatorTabs/index.js b/lib/theme/DiscriminatorTabs/index.js
index f5365082b3b04d779abbaf9e1c096586d8326881..91ef60b6d4b0e3d209ba2091722244b37b9fba02 100644
--- a/lib/theme/DiscriminatorTabs/index.js
+++ b/lib/theme/DiscriminatorTabs/index.js
@@ -228,7 +228,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   return react_1.default.createElement(
     "div",
     { className: "openapi-tabs__container" },
diff --git a/lib/theme/MimeTabs/index.js b/lib/theme/MimeTabs/index.js
index 93c055908069e6b5e62036a79a43d6e0cf162c7e..239c5eba10fd3e5f56213724f317c88c900b5bc4 100644
--- a/lib/theme/MimeTabs/index.js
+++ b/lib/theme/MimeTabs/index.js
@@ -261,7 +261,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   return react_1.default.createElement(
     "div",
     { className: "tabs-container" },
diff --git a/lib/theme/OperationTabs/index.js b/lib/theme/OperationTabs/index.js
index 6c7c7950d57234105392ef92db0c4816e96e5c69..dd951e519124c77cf5e84487845b5a45af803d3b 100644
--- a/lib/theme/OperationTabs/index.js
+++ b/lib/theme/OperationTabs/index.js
@@ -223,7 +223,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   return react_1.default.createElement(
     "div",
     { className: "tabs-container" },
diff --git a/lib/theme/SchemaTabs/index.js b/lib/theme/SchemaTabs/index.js
index d1924da270d632278ba05fbf811428290be949ac..e7123ef901d97fea2c30f6dd5930b26906a9bc35 100644
--- a/lib/theme/SchemaTabs/index.js
+++ b/lib/theme/SchemaTabs/index.js
@@ -242,7 +242,7 @@ function TabContent({ lazy, children, selectedValue }) {
   );
 }
 function TabsComponent(props) {
-  const tabs = (0, internal_1.useTabs)(props);
+  const tabs = (0, internal_1.useTabsContextValue)(props);
   return react_1.default.createElement(
     "div",
     { className: "openapi-tabs__schema-container" },
diff --git a/src/theme/ApiExplorer/CodeTabs/index.tsx b/src/theme/ApiExplorer/CodeTabs/index.tsx
index 06cd307dc1296a7218d99c801ac392d609751326..5d9dbb66597b936b5a4e7e08ce21384b1afd9b48 100644
--- a/src/theme/ApiExplorer/CodeTabs/index.tsx
+++ b/src/theme/ApiExplorer/CodeTabs/index.tsx
@@ -11,7 +11,7 @@ import {
   sanitizeTabsChildren,
   type TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import useIsBrowser from "@docusaurus/useIsBrowser";
@@ -43,7 +43,7 @@ function TabList({
   selectedValue,
   selectValue,
   tabValues,
-}: CodeTabsProps & ReturnType<typeof useTabs>) {
+}: CodeTabsProps & ReturnType<typeof useTabsContextValue>) {
   const tabRefs = useRef<(HTMLLIElement | null)[]>([]);
   const tabsScrollContainerRef = useRef<any>(null);
   const { blockElementScrollPositionUntilNextRender } =
@@ -192,7 +192,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: CodeTabsProps & ReturnType<typeof useTabs>): React.JSX.Element | null {
+}: CodeTabsProps & ReturnType<typeof useTabsContextValue>): React.JSX.Element | null {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -219,7 +219,7 @@ function TabContent({
 }
 
 function TabsComponent(props: CodeTabsProps & Props): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   const { className } = props;
 
   return (
diff --git a/src/theme/ApiTabs/index.tsx b/src/theme/ApiTabs/index.tsx
index 7545069ae9ec5d5655c6e6c8360c4a685f0f10ef..4757ce624db527f04a91a4a0308c282eb598817d 100644
--- a/src/theme/ApiTabs/index.tsx
+++ b/src/theme/ApiTabs/index.tsx
@@ -17,7 +17,7 @@ import {
   sanitizeTabsChildren,
   TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import { translate } from "@docusaurus/Translate";
@@ -42,7 +42,7 @@ function TabList({
     message: "Responses",
   }),
   id = "responses",
-}: TabListProps & ReturnType<typeof useTabs>) {
+}: TabListProps & ReturnType<typeof useTabsContextValue>) {
   const tabRefs: (HTMLLIElement | null)[] = [];
   const { blockElementScrollPositionUntilNextRender } =
     useScrollPositionBlocker();
@@ -194,7 +194,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: TabProps & ReturnType<typeof useTabs>): React.JSX.Element | null {
+}: TabProps & ReturnType<typeof useTabsContextValue>): React.JSX.Element | null {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -220,7 +220,7 @@ function TabContent({
   );
 }
 function TabsComponent(props: TabListProps): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   return (
     <div className="openapi-tabs__container">
       <TabList {...props} {...tabs} />
diff --git a/src/theme/DiscriminatorTabs/index.tsx b/src/theme/DiscriminatorTabs/index.tsx
index 0bb7c72fd0f1c0c9964de252aa93259b3065bd44..7600fd77f47f9075b4986fb31b038d82b8d78571 100644
--- a/src/theme/DiscriminatorTabs/index.tsx
+++ b/src/theme/DiscriminatorTabs/index.tsx
@@ -17,7 +17,7 @@ import {
   sanitizeTabsChildren,
   TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import useIsBrowser from "@docusaurus/useIsBrowser";
@@ -30,7 +30,7 @@ function TabList({
   selectedValue,
   selectValue,
   tabValues,
-}: TabProps & ReturnType<typeof useTabs>) {
+}: TabProps & ReturnType<typeof useTabsContextValue>) {
   const tabRefs: (HTMLLIElement | null)[] = [];
   const { blockElementScrollPositionUntilNextRender } =
     useScrollPositionBlocker();
@@ -172,7 +172,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: TabProps & ReturnType<typeof useTabs>): React.JSX.Element | null {
+}: TabProps & ReturnType<typeof useTabsContextValue>): React.JSX.Element | null {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -199,7 +199,7 @@ function TabContent({
   );
 }
 function TabsComponent(props: TabProps): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   return (
     <div className="openapi-tabs__container">
       <TabList {...props} {...tabs} />
diff --git a/src/theme/MimeTabs/index.tsx b/src/theme/MimeTabs/index.tsx
index 750982b32d2fcc071343b6e48a34fd78c6ceca18..8d6b6889b597bf262fbc3b3498322b2d06dce99d 100644
--- a/src/theme/MimeTabs/index.tsx
+++ b/src/theme/MimeTabs/index.tsx
@@ -17,7 +17,7 @@ import {
   sanitizeTabsChildren,
   TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import useIsBrowser from "@docusaurus/useIsBrowser";
@@ -38,7 +38,7 @@ function TabList({
   selectValue,
   tabValues,
   schemaType,
-}: Props & TabProps & ReturnType<typeof useTabs>): React.JSX.Element {
+}: Props & TabProps & ReturnType<typeof useTabsContextValue>): React.JSX.Element {
   const tabRefs: (HTMLLIElement | null)[] = [];
   const { blockElementScrollPositionUntilNextRender } =
     useScrollPositionBlocker();
@@ -208,7 +208,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: Props & TabProps & ReturnType<typeof useTabs>) {
+}: Props & TabProps & ReturnType<typeof useTabsContextValue>) {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -234,7 +234,7 @@ function TabContent({
   );
 }
 function TabsComponent(props: Props & TabProps): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   return (
     <div className="tabs-container">
       <TabList {...props} {...tabs} />
diff --git a/src/theme/OperationTabs/index.tsx b/src/theme/OperationTabs/index.tsx
index a0557924bee306da2508e2556f28a52391e3bae0..a3fe5bdda53fae2f5f602a198631d2233edf0833 100644
--- a/src/theme/OperationTabs/index.tsx
+++ b/src/theme/OperationTabs/index.tsx
@@ -17,7 +17,7 @@ import {
   sanitizeTabsChildren,
   TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import useIsBrowser from "@docusaurus/useIsBrowser";
@@ -29,7 +29,7 @@ function TabList({
   selectedValue,
   selectValue,
   tabValues,
-}: TabProps & ReturnType<typeof useTabs>) {
+}: TabProps & ReturnType<typeof useTabsContextValue>) {
   const tabRefs: (HTMLLIElement | null)[] = [];
   const { blockElementScrollPositionUntilNextRender } =
     useScrollPositionBlocker();
@@ -171,7 +171,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: TabProps & ReturnType<typeof useTabs>): React.JSX.Element | null {
+}: TabProps & ReturnType<typeof useTabsContextValue>): React.JSX.Element | null {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -197,7 +197,7 @@ function TabContent({
   );
 }
 function TabsComponent(props: TabProps): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   return (
     <div className="tabs-container">
       <TabList {...props} {...tabs} />
diff --git a/src/theme/SchemaTabs/index.tsx b/src/theme/SchemaTabs/index.tsx
index b94d12942ac37f7db551fe939bafb1e7fc2a43f1..7eb5ba973a888a356a4175af3db299954f13cb27 100644
--- a/src/theme/SchemaTabs/index.tsx
+++ b/src/theme/SchemaTabs/index.tsx
@@ -18,7 +18,7 @@ import {
   sanitizeTabsChildren,
   TabProps,
   useScrollPositionBlocker,
-  useTabs,
+  useTabsContextValue,
 } from "@docusaurus/theme-common/internal";
 import { TabItemProps } from "@docusaurus/theme-common/lib/utils/tabsUtils";
 import useIsBrowser from "@docusaurus/useIsBrowser";
@@ -40,7 +40,7 @@ function TabList({
   selectValue,
   tabValues,
   onChange,
-}: SchemaTabsProps & ReturnType<typeof useTabs>) {
+}: SchemaTabsProps & ReturnType<typeof useTabsContextValue>) {
   const tabRefs: (HTMLLIElement | null)[] = [];
   const { blockElementScrollPositionUntilNextRender } =
     useScrollPositionBlocker();
@@ -188,7 +188,7 @@ function TabContent({
   lazy,
   children,
   selectedValue,
-}: SchemaTabsProps & ReturnType<typeof useTabs>) {
+}: SchemaTabsProps & ReturnType<typeof useTabsContextValue>) {
   const childTabs = (Array.isArray(children) ? children : [children]).filter(
     Boolean
   ) as ReactElement<TabItemProps>[];
@@ -215,7 +215,7 @@ function TabContent({
   );
 }
 function TabsComponent(props: SchemaTabsProps): React.JSX.Element {
-  const tabs = useTabs(props);
+  const tabs = useTabsContextValue(props);
   return (
     <div className="openapi-tabs__schema-container">
       <TabList {...props} {...tabs} />
